### PR TITLE
feat: add pagination for list resources

### PR DESCRIPTION
**Pull request by anjali9791**
Closes: https://github.com/goto/guardian/issues/76

### CODE REVIEW

File: api/handler/v1beta1/resource.go [hunk 1]
```diff
@@ -25,31 +26,64 @@ func (s *GRPCServer) ListResources(ctx context.Context, req *guardianv1beta1.Lis
...
+		}
+		total = totalRecord
+		return nil
+	})
+
+	if err := eg.Wait(); err != nil {
+		return nil, 0, err
+	}
+	var resourceProtos []*guardianv1beta1.Resource
+	for i, a := range resources {
```
Comment (bsushmith) [inline api/handler/v1beta1/resource.go:78]:
```suggestion
	for i, r := range resources {
```

File: api/handler/v1beta1/resource.go [hunk 2]
```diff
@@ -25,31 +26,64 @@ func (s *GRPCServer) ListResources(ctx context.Context, req *guardianv1beta1.Lis
...
+		resourceProto, err := s.adapter.ToResourceProto(resources[i])
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "failed to parse resource %v: %v", r.Name, err)
+			return nil, 0, status.Errorf(codes.Internal, "failed to parse resource %v: %v", a.Name, err)
```
Comment (bsushmith) [inline api/handler/v1beta1/resource.go:81]:
```suggestion
			return nil, 0, status.Errorf(codes.Internal, "failed to parse resource %v: %v", r.Name, err)
```

File: internal/store/postgres/resource_repository_test.go [hunk 3]
```diff
@@ -225,6 +225,14 @@ func (s *ResourceRepositoryTestSuite) TestGetOne() {
...
 }
 
+func (s *ResourceRepositoryTestSuite) TestGetResourcesTotalCount() {
+	s.Run("should return 0", func() {
+		_, actualError := s.repository.GetResourcesTotalCount(context.Background(), domain.ListResourcesFilter{})
+
+		s.Nil(actualError)
+	})
+}
+
```
Comment (bsushmith) [inline internal/store/postgres/resource_repository_test.go:235]: Can you also add a positive test case for `size` and `offset` changes?

File: api/handler/v1beta1/resource.go (hunk 1 above)
Comment (anjali9791) [reply api/handler/v1beta1/resource.go]: done

File: internal/store/postgres/resource_repository_test.go (hunk 3 above)
Comment (anjali9791) [reply internal/store/postgres/resource_repository_test.go:250]: done

File: api/handler/v1beta1/resource.go (hunk 2 above)
Comment (anjali9791) [reply api/handler/v1beta1/resource.go]: done

**Review by bsushmith — APPROVED** [review]

**Pull request merged**